src/java/org/apache/cassandra/db/guardrails/GuardrailsMBean.java

```diff
@@ -651,4 +651,56 @@
      * @param value {@code true} if 0 default TTL on TWCS tables is allowed, {@code false} otherwise.
      */
     void setZeroTTLOnTWCSEnabled(boolean value);
+
+    /**
+     * @param currentTimestamp the current timestamp
+     * @return The highest acceptable timestamp before triggering a warning
+     */
+    long getMaximumTimestampWarnThreshold(long currentTimestamp);
+
+    /**
+     * Sets the warning upper bound for user supplied timestamps
+     *
+     * @param durationSeconds currentTimestamp + TimeUnit.SECONDS.toMicros(duration)
+     */
+    void setMaximumTimestampWarnThreshold(int durationSeconds);
```

Comment:
During the initial review for guardrails we decided to use a common setter method for both the warn and fail threshold. IIRC the reason was that the requisite of keeping the warn threshold lesser than the fail threshold imposes a certain order on how the separate setter would need to be called. For example, consider the following:
```java
guardrails().setMinimumTimestampWarnThreshold(1);
guardrails().setMinimumTimestampFailThreshold(2);
```
That might fail or not fail depending on the previous value of the fail threshold. However, the combined setter would always succeed:
```java
guardrails().setMinimumTimestampThreshold(1, 2);
```
I think we should keep the current style of combined setters in `GuardrailsMBean` for the sake of consistency.

Reply:
Will change to use a single setter